### PR TITLE
[fcos] pkg/controller/common/helpers: replace duplicate files in ConvertIgnition2to3

### PR DESCRIPTION
**Pull request by vrutkovs**
When Ignition2 files are converted to Ignition3 we have to ensure no files are duplicated. This commit would rewrite previous versions of the file with the latest content.

Fixes https://github.com/openshift/okd/issues/237

### CODE REVIEW

File: pkg/controller/common/helpers.go [hunk 1]
```diff
@@ -125,13 +125,31 @@ func WriteTerminationError(err error) {
...
 }
 
+func replaceOrAppend(files []ign2types.File, file ign2types.File) {
+	for i, f := range files {
+		if f.Node.Path == file.Node.Path {
+			files[i] = file
+			return
+		}
+	}
+	files = append(files, file)
```
Comment (LorbusChris) [inline pkg/controller/common/helpers.go:135]: I think we're missing a `return` after this line

Comment (LorbusChris) [reply pkg/controller/common/helpers.go:135]: actually, it needs look like this: https://github.com/openshift/machine-config-operator/pull/1703/commits/db4a802817e7339da5ae74d751069e1d3e67614d
anoalog to the way it's done in installer

Comment (vrutkovs) [reply pkg/controller/common/helpers.go]: I was mutating `files` there, reworked to return the array

**Comment by vrutkovs** [comment]
`failed to acquire lease: resources not found`

/retest

**Comment by LorbusChris** [comment]
@vrutkovs compare with https://github.com/coreos/ign-converter/pull/9

**Comment by vrutkovs** [comment]
I'd prefer ignconverter solution in the long term, however this patch would unblock users using mirroring quickier

**Pull request merged**